middleware/http_tracing.go

```diff
@@ -54,3 +54,19 @@ func ExtractTraceID(ctx context.Context) (string, bool) {
 
 	return sctx.TraceID().String(), true
 }
+
+// ExtractSampledTraceID works like ExtractTraceID but the returned bool is only
+// true if the returned trace id is both sampled and the sampling decision has
+// been finalized.
+func ExtractSampledTraceID(ctx context.Context) (string, bool) {
+	sp := opentracing.SpanFromContext(ctx)
+	if sp == nil {
+		return "", false
+	}
+	sctx, ok := sp.Context().(jaeger.SpanContext)
+	if !ok {
+		return "", false
+	}
+
+	return sctx.TraceID().String(), sctx.IsSampled() && sctx.IsSamplingFinalized()
```

Comment:
I would drop `IsSamplingFinalized()`.  The code paths that set sampling "finalized" or not are complex.  I think for almost all scenarios just `IsSampled()` is sufficient and less prone to complex edge cases.

Reply:
If you say so, happy to do so.